### PR TITLE
Docs: Custom `babel-plugin-espower` patterns

### PR DESCRIPTION
**Pull request by trbrc**
Suggestion for documentation of solution to #2132 

Related discussion in #1255 suggests that I'm not the only one looking for how to do this, so might as well add it to the documentation.

### CODE REVIEW

File: docs/recipes/babel.md [hunk 1]
```diff
@@ -101,6 +101,31 @@ Note that this *does not* stop AVA from compiling your test files using Babel.
...
+```json
+{
+	"ava": {
+		"babel": {
+			"testOptions": {
+				"plugins": [
+					["babel-plugin-espower", {
+						"patterns": [
+							"assert(value, [message])",
+							"test.assert(value, [message])"
```
Comment (novemberborn) [inline docs/recipes/babel.md]: This would be `t.assert`.

**Comment by novemberborn** [comment]
I wonder if, now that only `t.assert()` has Power Assert applied to it, we can't transform `assert` out of the box. I'd rather not document these approaches because it both implies they're supported *and* that they're necessary.

@sindresorhus what do you think?

File: docs/recipes/babel.md (hunk 1 above)
Comment (trbrc) [reply docs/recipes/babel.md]: My intention was to show how to deal with an argument name other than `t`, and I thought `test` ought to be the most clear example. Perhaps the example should include `t.assert` as well.

**Comment by trbrc** [comment]
Regardless of varying opinions about how AVA should be used, it’s simply unexpected that your tests behave a little different because you name an argument something else. I don’t like surprises in my tests. I believe it’s a good thing to include somewhere in the documentation, to at least make it a little less surprising. If it can be made more robust out of the box, that would be even better though.

File: docs/recipes/babel.md (hunk 1 above)
Comment (novemberborn) [reply docs/recipes/babel.md]: It's this confusion that makes me think we should support `assert` usage by default.

**Comment by novemberborn** [comment]
> it’s simply unexpected that your tests behave a little different because you name an argument something else

The tests pass or fail the same though, it's just the extra information that's no longer provided. I think people end up disabling that enhancement fairly often too.

File: docs/recipes/babel.md (hunk 1 above)
Comment (sindresorhus) [reply docs/recipes/babel.md]: I was confused by `test.assert` too. I think you should use a totally unrelated name like `foo.assert`.

**Comment by novemberborn** [comment]
> I wonder if, now that only `t.assert()` has Power Assert applied to it, we can't transform `assert` out of the box. I'd rather not document these approaches because it both implies they're supported _and_ that they're necessary.

Let's apply it to anything matching `<identifier>.assert()` and `assert()` itself.

**Comment by novemberborn** [comment]
See https://github.com/avajs/babel-preset-transform-test-files/pull/18.

**Pull request closed without merging**